intellij/src/saros/intellij/filesystem/IntellijResourceImplV2.java

```diff
@@ -0,0 +1,130 @@
+package saros.intellij.filesystem;
+
+import static saros.filesystem.IResource.Type.FOLDER;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import saros.filesystem.IContainer;
+import saros.filesystem.IPath;
+import saros.filesystem.IProject;
+import saros.filesystem.IResource;
+import saros.intellij.editor.ProjectAPI;
+
+/** Abstract Intellij implementation of the Saros resource interface. */
+// TODO rename to AbstractIntellijResource
+public abstract class IntellijResourceImplV2 implements IResource {
+
+  /** Reference point to which this resource belongs. */
+  protected final IntellijReferencePointImpl referencePoint;
+
+  /** Relative path from the given reference point. */
+  protected final IPath relativePath;
+
+  public IntellijResourceImplV2(
+      @NotNull IntellijReferencePointImpl referencePoint, @NotNull IPath relativePath) {
+
+    if (relativePath.isAbsolute()) {
+      throw new IllegalArgumentException("The given path must not be absolute - " + relativePath);
+    }
+
+    this.referencePoint = referencePoint;
+    this.relativePath = relativePath;
+  }
+
+  /**
+   * Returns the virtual file represented by this resource.
+   *
+   * @return the virtual file represented by this resource or <code>null</code> if no such resource
+   *     exists in the local VFS snapshot
+   */
+  @Nullable
+  protected VirtualFile getVirtualFile() {
+    return referencePoint.findVirtualFile(relativePath);
+  }
+
+  @Override
+  @NotNull
+  public String getName() {
+    return relativePath.lastSegment();
```

Comment:
Returns `null` if relative path is empty. Either forbid resources with empty paths (and adjust `VirtualFileConverter` to reflect this) or use virtual file to obtain name.

Reply:
Fixed. Given paths must now not be empty.